Blockzilla/OverlayView.swift

```diff
@@ -9,103 +9,161 @@ protocol OverlayViewDelegate: class {
         }
-
         searchBorder.snp.makeConstraints { make in
             make.leading.trailing.equalTo(self)
             make.top.equalTo(searchButton.snp.bottom)
             make.height.equalTo(1)
+            
         }
-
+        
+        copyButton.setImage(#imageLiteral(resourceName: "icon_paste"), for: .normal)
+        copyButton.setImage(#imageLiteral(resourceName: "icon_paste"), for: .highlighted)
+        copyButton.titleLabel?.font = UIConstants.fonts.copyButton
+        setUpOverlayButton(button: copyButton)
+        copyButton.addTarget(self, action: #selector(didPressCopy), for: .touchUpInside)
+        addSubview(copyButton)
+        
+        copyBorder.backgroundColor = UIConstants.colors.copyButtonBorder
+        addSubview(copyBorder)
+        
+        copyButton.snp.makeConstraints { make in
+            make.top.leading.trailing.equalTo(self)
+            
+        }
+        copyBorder.snp.makeConstraints { make in
+            make.leading.trailing.equalTo(self)
+            make.top.equalTo(copyButton.snp.bottom)
+            make.height.equalTo(1)
+        }
+        
+        settingsButton.setImage(#imageLiteral(resourceName: "icon_settings"), for: .normal)
+        settingsButton.contentEdgeInsets = UIEdgeInsets(top: 20, left: 20, bottom: 20, right: 20)
+        settingsButton.addTarget(self, action: #selector(didPressSettings), for: .touchUpInside)
+        settingsButton.accessibilityLabel = UIConstants.strings.browserSettings
+        addSubview(settingsButton)
+        
         settingsButton.snp.makeConstraints { make in
             make.trailing.equalTo(self)
             bottomConstraint = make.bottom.equalTo(self).constraint
         }
     }
-
     required init?(coder aDecoder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
+    func setUpOverlayButton (button: InsetButton) {
+        button.titleLabel?.lineBreakMode = .byTruncatingTail
+        if UIView.userInterfaceLayoutDirection(for: button.semanticContentAttribute) == .rightToLeft {
+            button.contentHorizontalAlignment = .right
+        } else {
+            button.contentHorizontalAlignment = .left
+        }
+        
+        let padding = UIConstants.layout.searchButtonInset
+        button.imageEdgeInsets = UIEdgeInsets(top: padding, left: padding, bottom: padding, right: padding)
+        if UIView.userInterfaceLayoutDirection(for: button.semanticContentAttribute) == .rightToLeft {
+            button.titleEdgeInsets = UIEdgeInsets(top: padding, left: padding, bottom: padding, right: padding * 2)
+        } else {
+            button.titleEdgeInsets = UIEdgeInsets(top: padding, left: padding * 2, bottom: padding, right: padding)
+        }
+    }
+    
+    func verifyUrl (urlString: String?) -> Bool {
+        if let urlString = urlString {
+            if let url = URL(string: urlString) {
+                return UIApplication.shared.canOpenURL(url)
```

Comment:
I don't think this is correct. In this particular case, aren't you just trying to find out if the URL is `http` or `https`? If so, you should just test for that. I recently learned that `canOpenURL()` has some intentional limitations requiring that the application whitelist all possible URL schemes you plan on testing for. After 50 calls, this will always return `false`: https://developer.apple.com/documentation/uikit/uiapplication/1622952-canopenurl